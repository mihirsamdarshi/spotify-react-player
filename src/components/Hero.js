import React, { useContext, useEffect } from 'react';
import '../stylesheets/Hero.scss';
import { isEmpty } from 'lodash';
import { getHashParams, SetLoginDispatch } from '../scripts/helpers';

const authEndpoint = 'https://accounts.spotify.com/authorize';
const clientId = "ccc1ea71746a4d4ea8468a7cc7d1d9f0";
const redirectUri = "http://localhost:3000";
const scopes = [
    "user-read-private",
    "user-read-playback-state",
    "user-modify-playback-state",
    "streaming",
    "user-read-currently-playing",
];

const Hero = props => {
    const setAccessToken = useContext(SetLoginDispatch);

    useEffect(() => {
        const token = getHashParams();
        if (!isEmpty(token)) {
            setAccessToken(token);
        }
    }, [setAccessToken]);

    return (
        <div className="fillWindow">
            <div className="heroInfo">
                <h1> Applotify </h1>
                <h3> Play Spotify Playlists&#42; </h3>
            </div>
            <div className="centeredBtn">
                <a
                    className="spotifyLoginBtn"
                    href={`${authEndpoint}?response_type=token&client_id=${clientId}&redirect_uri=${redirectUri}&scope=${scopes.join("%20")}&show_dialog=true`}
                >
                    Login With Spotify
                </a>
            </div>

            <p className="asterisk"> &#42; only available in Chrome and Firefox browsers and to users with Spotify Premium due to Spotify API limitations</p>
        </div>
    )
};

export default Hero;
