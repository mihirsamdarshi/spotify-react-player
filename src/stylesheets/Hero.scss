.fillWindow {
  height: 100%;
  align-items: center;
}

.heroInfo {
  text-align: center;
}

.centeredBtn {
  margin: 4% auto;
  width: 25%;
}

.spotifyLoginBtn {
  font-size: 30px;
  border-radius: 500px;
  padding: 19px 56px 21px;
  color: #FFF;
  background-color: #1DB954;
  min-width: 160px;
  font-weight: 700;
  text-align: center;
  border: 1px solid transparent;
}

.asterisk {
  text-align: center;
  font-size: 0.5em;
  margin-top: 20%;
}
