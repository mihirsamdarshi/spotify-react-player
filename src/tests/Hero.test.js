import ReactDOM from 'react-dom';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import { isEmpty } from 'lodash';

import Hero from '../components/Hero';
import { getHashParams } from '../scripts/helpers';


it('renders the Hero component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Hero />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should initialize with text', () => {
    const component = TestRenderer.create(<Hero />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

describe('the hero component', () => {
    let div;
    beforeEach(() => {
        div = document.createElement('div');
        ReactTestUtils.act(() => {
            ReactDOM.render(<Hero />, div)
        })
    });
    afterEach(() => ReactDOM.unmountComponentAtNode(div));

    it('should have an h1 tag', () => {
        const titleHeader = div.querySelector('h1')
        expect(titleHeader).toBeDefined();
    });

    it('should have an h3 tag', () => {
        const subtitleHeader = div.querySelector('h3')
        expect(subtitleHeader).toBeDefined();
    });

    it('should have a p tag', () => {
        const disclaimer = div.querySelector('p')
        expect(disclaimer).toBeDefined();
    });

    describe('the button', () => {
        it('should redirect upon click', () => {
            const button = div.querySelector('a');
            ReactTestUtils.act(() => {
                ReactTestUtils.Simulate.click(button);
            })
        });
    });

    it('should return null for hash params', () => {
        expect(isEmpty(getHashParams())).toBe(true);
    });

    it('should return a value for hash params', () => {
        const output = {"access_token": "helloworld", "expires_in": "69", "token_type": "Bearer"};
        global.window = Object.create(window);
        const url = "#access_token=helloworld&token_type=Bearer&expires_in=69";
        Object.defineProperty(window, "location", {
            value: {
                hash: url
            },
            writable: true
        });
        expect(getHashParams()).toStrictEqual(output);
    });
});
