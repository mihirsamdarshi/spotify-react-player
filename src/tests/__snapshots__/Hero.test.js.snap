// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should initialize with text 1`] = `
<div
  className="fillWindow"
>
  <div
    className="heroInfo"
  >
    <h1>
       Applotify 
    </h1>
    <h3>
       Play Spotify Playlists* 
    </h3>
  </div>
  <div
    className="centeredBtn"
  >
    <a
      className="spotifyLoginBtn"
      href="https://accounts.spotify.com/authorize?response_type=token&client_id=ccc1ea71746a4d4ea8468a7cc7d1d9f0&redirect_uri=http://localhost:3000&scope=user-read-private%20user-read-playback-state%20user-modify-playback-state%20streaming%20user-read-currently-playing&show_dialog=true"
    >
      Login With Spotify
    </a>
  </div>
  <p
    className="asterisk"
  >
     * only available in Chrome and Firefox browsers and to users with Spotify Premium due to Spotify API limitations
  </p>
</div>
`;
